### PR TITLE
feat: support http and https url in custom linters

### PR DESCRIPTION
**Pull request by draveness**
Feature Support: Custom linters can download and open dynamic libraries with HTTP and HTTPs.

### CODE REVIEW

File: pkg/lint/lintersdb/manager.go [hunk 1]
```diff
@@ -426,3 +451,26 @@ func (m Manager) getAnalyzerPlugin(path string) (AnalyzerPlugin, error) {
...
 	return analyzerPlugin, nil
 }
+
+func downloadFile(path string) (string, error) {
+	temfile, err := ioutil.TempFile("", "plugin")
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := http.Get(path)
```
Comment (sayboras) [inline pkg/lint/lintersdb/manager.go:461]: please correct lint error here

File: pkg/lint/lintersdb/manager.go [hunk 2]
```diff
@@ -409,6 +413,27 @@ type AnalyzerPlugin interface {
...
+	pathURL, err := url.Parse(path)
+	if err != nil {
+		return nil, fmt.Errorf("plugin %s has invalid path", path)
+	}
+
+	switch pathURL.Scheme {
+	case "http", "https":
+		pluginPath, downloadErr := downloadFile(path)
+		if downloadErr != nil {
+			return nil, fmt.Errorf("download plugin %s error, err: %s", path, err.Error())
```
Comment (sayboras) [inline pkg/lint/lintersdb/manager.go:425]: %w for error to wrap it

File: pkg/lint/lintersdb/manager.go [hunk 3]
```diff
@@ -409,6 +413,27 @@ type AnalyzerPlugin interface {
 }
 
 func (m Manager) getAnalyzerPlugin(path string) (AnalyzerPlugin, error) {
+	pathURL, err := url.Parse(path)
+	if err != nil {
+		return nil, fmt.Errorf("plugin %s has invalid path", path)
```
Comment (sayboras) [inline pkg/lint/lintersdb/manager.go:418]: if `path` is not valid path (e.g. maybe just local file system), will it fail and return here right away ?

**Review by sayboras — COMMENTED** [review]
Seems like this PR is still WIP, I am marking it as draft. Feel free to mark it as ready once all CI jobs are passed.

File: pkg/lint/lintersdb/manager.go (hunk 3 above)
Comment (draveness) [reply pkg/lint/lintersdb/manager.go:418]: by valid path, what do you mean here? maybe you can refer to https://play.golang.org/p/d5poeeQM8US

File: pkg/lint/lintersdb/manager.go (hunk 1 above)
Comment (draveness) [reply pkg/lint/lintersdb/manager.go]: add a nolint here

**Comment by draveness** [comment]
could you help to take a look at the failed CI? i don't think it's related to this change. @sayboras

**Comment by sayboras** [comment]
> could you help to take a look at the failed CI? i don't think it's related to this change.

Please rebase with master to fix check_generated failure. Other jobs are passing now. Feel free to send for review once you think it's ready.

PS: Some description in PR will be nice, and make it easier for reviewer.

**Review by blaquez — APPROVED** [review]

**Review by blaquez — APPROVED** [review]

**Review by blaquez — APPROVED** [review]

**Review by blaquez — APPROVED** [review]

**Review by blaquez — APPROVED** [review]

**Comment by ernado** [comment]
1. Is it safe to allow downloading code in such way by default? Looks like there will be no validation, e.g. no checksums. I think it would be better to add some feature flag to enable this logic.
2. Probably this can be improved via caching (we already have golangci lint cache) and retries, but current implementation is fine as experiment.

**Pull request closed without merging**